commit 2a731a6951d287b16585a8a1e9a1fc14b6569d04 (refs/changes/41/841/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2016-11-03T13:14:40+01:00 (2 years, 11 months ago)
    
    TOOLS-1559 wrap sapiadm and manta-login to avoid RBAC break

diff --git a/scripts/headnode.sh b/scripts/headnode.sh
index 2c27d7d4..f60bab16 100755
--- a/scripts/headnode.sh
+++ b/scripts/headnode.sh
@@ -788,13 +788,17 @@ if setup_state_not_seen "sdczones_created"; then
         mv ${file} /opt/smartdc/bin/${tool}
     done
 
-    # Copy sapiadm into the GZ from the SAPI zone.
+    # Create wrapper for sapiadm into the GZ and execute it from the SAPI zone.
     zone_uuid=$(vmadm lookup tags.smartdc_role=sapi | head -n 1)
     if [[ -n ${zone_uuid} ]]; then
-        from_dir=/zones/${zone_uuid}/root/opt/smartdc/config-agent/cmd
+        from_dir=/zones/${zone_uuid}/root/opt/smartdc
         to_dir=/opt/smartdc/bin
         rm -f ${to_dir}/sapiadm
-        ln -s ${from_dir}/sapiadm.js ${to_dir}/sapiadm
+        cat <<-EOF > ${to_dir}/sapiadm
+#!/bin/bash
+exec ${from_dir}/sapi/build/node/bin/node ${from_dir}/config-agent/cmd/sapiadm.js "\$@"
+EOF
+        chmod +x ${to_dir}/sapiadm
     fi
 
     setup_state_add "sdczones_created"
diff --git a/scripts/setup_manta_zone.sh b/scripts/setup_manta_zone.sh
index 90a5ad18..40df66e0 100755
--- a/scripts/setup_manta_zone.sh
+++ b/scripts/setup_manta_zone.sh
@@ -169,9 +169,14 @@ function copy_manta_tools {
         rm -f ${to_dir}/manta-oneach
 
         mkdir -p /opt/smartdc/manta-deployment/log
-        # manta-login is a bash script, so we can link it directly.
-        ln -s ${from_dir}/bin/manta-login ${to_dir}/manta-login
 
+        # while manta-login is a bash script and we could link it directly,
+        # we are using a little wrapper to avoid permission issues on the GZ
+        cat <<-EOF > ${to_dir}/manta-login
+#!/bin/bash
+exec ${from_dir}/bin/manta-login "\$@"
+EOF
+        chmod +x ${to_dir}/manta-login
         #
         # manta-adm and manta-oneach are node programs, so we must write little
         # wrappers that call the real version using the node delivered in the
